api/v1/nodescan/service.go

```diff
@@ -227,6 +227,12 @@ func (s *serviceImpl) GetNodeVulnerabilities(_ context.Context, req *v1.GetNodeV
 
 	resp.OperatingSystem, resp.KernelVulnerabilities, resp.KernelComponent, err = s.evaluateLinuxKernelVulns(req)
 	if err != nil {
+		// If the node is unsupported, exit early.
+		if err == kernelparser.ErrNodeUnsupported {
+			resp.Notes = append(resp.Notes, v1.NodeNote_NODE_OS_UNSUPPORTED)
+			return resp, nil
+		}
```

Comment:
Why a "note" and not a plain empty response, or an error? Does the UI consume empty responses with notes in a meaningful way?

I noticed that we return empty responses in some scenarios (like "windows"). I would instead use the same pattern unless there is a reasonable tradeoff because adding a specialized error increases complexity, and I want to avoid it.

Reply:
An error implies something went wrong when scanning. Putting a note here tells us that scanning went exactly as expected, but the results are incomplete do to the reasons shown in the note. We have the same idea for images: https://github.com/stackrox/scanner/blob/master/proto/scanner/api/v1/note.proto#L9. The UI displays the scanning notes in the Vuln Mgmt pages. See https://github.com/stackrox/stackrox/pull/1275 for the related PR.

Good idea adding a note for something like "windows". Will add that

Reply:
It makes sense, considering the UI handles notes in the way you clarified.

About incomplete results and how best we can report them (as errors or not), that's debatable. It is better explained as a convention rather than a principle, and I am unsure if you are pointing out a convention. But, considering the notes are handled the way they are, I am OK with your approach anyway.